[Frontend] add chunking audio for > 30s audio

nguyenhoangthuan99 opened the pull request:
## Purpose
fix #15012 

My approach is to cut the lowest energy point of the last 1 second of each 30s audio chunk, then add all chunks to the engine client. wdyt @NickLucche 

## Test Plan
using openai python sdk to test both 
```
from openai import OpenAI
import os
client = OpenAI(api_key="key",base_url="http://localhost:8000/v1") 

def transcribe_audio(file_path):
    with open(file_path, "rb") as audio_file:
        transcription = client.audio.transcriptions.create(
            model="whisper-small", 
            file=audio_file
        )
        return transcription.text
def transcribe_audio_stream(file_path):
    with open(file_path, "rb") as audio_file:
        transcription = client.audio.transcriptions.create(
            model="whisper-small", 
            file=audio_file,
            stream = True
        )
        for event in transcription:
            print(event)
        return None
# Usage
result = transcribe_audio("path/to/>30s/audio")
print("Transcription:", result)

transcribe_audio_stream("path/to/>30s/audio")
```
## Test Result
The transcription text is printed successfully

## (Optional) Documentation Update

<!--- pyml disable-next-line no-emphasis-as-heading -->

nguyenhoangthuan99 committed:
[Frontend] add chunking audio for > 30s audio

Signed-off-by: nguyenhoangthuan99 <thuanhppro12@gmail.com>

nguyenhoangthuan99 committed:
[Frontend] fix linting and pre commit check

Signed-off-by: nguyenhoangthuan99 <thuanhppro12@gmail.com>

nguyenhoangthuan99 committed:
[Frontend] fix linting and pre commit check

Signed-off-by: nguyenhoangthuan99 <thuanhppro12@gmail.com>

DarkLight1337 commented:
cc @NickLucche

NickLucche reviewed (changes requested):
Hey, thanks a lot for contributing to vLLM!

Good job here, although the solution isn't super efficient (eg no batching on pre-processing) I think it will work very well. 
There's also room for optimization in a separate PR for the splitting code: we could vectorize the RMS computation better with numpy or at least give numba a try to try and reduce that overhead.

What we need to add in this PR though is at least a test case with a longer audio (splitting code testing would be nice but more optional).
To do that we need to add a new asset to the s3 bucket. @DarkLight1337 could you help here?
I've been using [this one](https://www.archive.org/download/multilingual_short_works_013_1507_librivox/msw013_13azacinto_foscolo_mer_128kb.mp3) to test, but any audio would do. LibriVox ones are public https://librivox.org/.

NickLucche commented:
Also, this test will fail now https://github.com/vllm-project/vllm/blob/main/tests/entrypoints/openai/test_transcription_validation.py#L77, we need to remove it

nguyenhoangthuan99 committed:
[Frontend] fix comment

Signed-off-by: nguyenhoangthuan99 <thuanhppro12@gmail.com>

nguyenhoangthuan99 committed:
[Frontend] remove bad request test and add long audio test

Signed-off-by: nguyenhoangthuan99 <thuanhppro12@gmail.com>

nguyenhoangthuan99 committed:
[Frontend] fix precommit hook fail

Signed-off-by: nguyenhoangthuan99 <thuanhppro12@gmail.com>

nguyenhoangthuan99 committed:
[Frontend] fix precommit hook fail

Signed-off-by: nguyenhoangthuan99 <thuanhppro12@gmail.com>

nguyenhoangthuan99 commented:
Hi @NickLucche , I fixed all your comments and added 1 test for long audio request - for this test, I just duplicate the `marry_had_lamb` audio 10 times and expect the string `Mary had a little lamb` appears 10 times in the transcription result. Can you take a look at it.

NickLucche reviewed (approved):
LGTM! Thanks for re-using the same test, I had forgotten I had one with a repeated timeseries.

DarkLight1337 reviewed (approved):
LGTM as well

nguyenhoangthuan99 commented:
Hi @NickLucche I see all checks are passed, can we merge this PR? 🙏
